qa/restart-upgrade/src/test/java/org/opensearch/knn/bwc/IndexingIT.java

```diff
@@ -0,0 +1,35 @@
+/*
+ *  Copyright OpenSearch Contributors
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.bwc;
+
+import static org.opensearch.knn.TestUtils.KNN_BWC_PREFIX;
+
+public class IndexingIT extends AbstractRestartUpgradeTestCase {
+    private final String CLASS_NAME = "indexingit";
+    private final String testIndexName = KNN_BWC_PREFIX + CLASS_NAME + "test-index";
+    private final String testFieldName = "test-field";
+    private final int dimensions = 5;
+    private final int k = 5;
+    private final int addDocsCnt = 10;
+
+    public void testKnnDefaultIndexSettings() throws Exception {
+        waitForClusterHealthGreen("3");
+
+        if (isRunningAgainstOldCluster()) {
+            createKnnIndex(testIndexName, getKNNDefaultIndexSettings(), createKnnIndexMapping(testFieldName, dimensions));
+            addKnnDocsBWCTests(testIndexName, testFieldName, dimensions, 0, addDocsCnt);
+        }
+
+        else {
```

Comment:
Why not change it into two tests? 1 for old clusters and other for otherwise ( use skip if required )

Reply:
By using setIndex() to create test indices, we cannot split the tests into 2 separate test methods as if the method name changes the index name changes and we need same index name for a given test in both old BWC version and after upgrading to latest version.